tests/system/test_client.py

```diff
@@ -1052,6 +1052,211 @@ def test_load_table_from_file_w_explicit_location(self):
+            "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/b-twitter.avro",
+            "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/c-twitter.avro",
+        ]
+
+        # Because referenceFileSchemaUri is set as a-twitter, the table will have a-twitter schema
+        reference_file_schema_uri = "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/a-twitter.avro"
+
+        # Create external data configuration
+        external_config = bigquery.ExternalConfig(bigquery.ExternalSourceFormat.AVRO)
+        external_config.source_uris = source_uris
+        external_config.reference_file_schema_uri = reference_file_schema_uri
+
+        table = bigquery.Table(table_ref)
+        table.external_data_configuration = external_config
+
+        table = client.create_table(table)
+
+        # Get table created by the create_table API call
+        generated_table = client.get_table(table_ref)
+
+        self.assertEqual(generated_table.schema, expected_schema)
+        self.assertEqual(
+            generated_table.external_data_configuration._properties[
+                "referenceFileSchemaUri"
+            ],
+            reference_file_schema_uri,
+        )
+
+        # Clean up test
+        self.to_delete.insert(0, generated_table)
+
+    def test_load_table_from_uri_with_reference_file_schema_uri_avro(self):
+        dataset_id = _make_dataset_id("test_reference_file_avro")
+        self.temp_dataset(dataset_id)
+        client = Config.CLIENT
+        dataset_ref = bigquery.DatasetReference(client.project, dataset_id)
+        table_id = "test_ref_file_avro"
+        table_ref = bigquery.TableReference(dataset_ref=dataset_ref, table_id=table_id)
+
+        expected_schema = [
+            bigquery.SchemaField("username", "STRING", mode="NULLABLE"),
+            bigquery.SchemaField("tweet", "STRING", mode="NULLABLE"),
+            bigquery.SchemaField("timestamp", "STRING", mode="NULLABLE"),
+            bigquery.SchemaField("likes", "INTEGER", mode="NULLABLE"),
+        ]
+
+        # By default, the table should have the c-twitter schema because it is lexicographically last:
+        # a-twitter schema: (username, tweet, timestamp, likes)
+        # b-twitter schema: (username, tweet, timestamp)
+        # c-twitter schema: (username, tweet)
+        source_uris = [
+            "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/a-twitter.avro",
+            "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/b-twitter.avro",
+            "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/c-twitter.avro",
+        ]
+
+        # Because referenceFileSchemaUri is set as a-twitter, the table will have a-twitter schema
+        reference_file_schema_uri = "gs://cloud-samples-data/bigquery/federated-formats-reference-file-schema/a-twitter.avro"
```

Comment:
same non blocking nit about variables

Reply:
tbh, I am torn about this because it might make it slightly less readable? Not sure I want to die on this hill